presto-native-execution/presto_cpp/main/common/ConfigReader.cpp

```diff
@@ -45,7 +70,9 @@ std::unordered_map<std::string, std::string> readConfig(
     const auto name = line.substr(0, delimiterPos);
     VELOX_CHECK(!name.empty(), "property pair '{}' has empty key", line);
     const auto value = line.substr(delimiterPos + 1);
-    properties.emplace(name, value);
+    const auto adjustedValue = extractValueIfEnvironmentVariable(value);
+    adjustedValue.has_value() ? properties.emplace(name, adjustedValue.value())
```

Comment:
During our sync, I misunderstood that we are redundantly updating non-env values.
How about we use `void extractValueIfEnvironmentVariable(std::string& value)`?
We can then use 
```
extractValueIfEnvironmentVariable(value);
properties.emplace(name, value);
```

Reply:
Yes. We used to copy it around if the value was a real value and not envvar.
This approach also works and might be a bit cleaner.